src/sentry/event_manager.py

```diff
@@ -860,31 +877,36 @@ def save(self, project_id, raw=False, assume_normalized=False):
             #         group.id: 1,
             #     },
             # })
-            (tsdb.models.frequent_environments_by_group, {
-                group.id: {
-                    environment.id: 1,
-                },
-            })
         ]
 
-        if release:
+        if group:
             frequencies.append(
-                (tsdb.models.frequent_releases_by_group, {
+                (tsdb.models.frequent_environments_by_group, {
                     group.id: {
-                        grouprelease.id: 1,
+                        environment.id: 1,
                     },
                 })
             )
 
-        tsdb.record_frequency_multi(frequencies, timestamp=event.datetime)
+            if release:
+                frequencies.append(
+                    (tsdb.models.frequent_releases_by_group, {
+                        group.id: {
+                            grouprelease.id: 1,
+                        },
+                    })
+                )
 
-        UserReport.objects.filter(
-            project=project,
-            event_id=event_id,
-        ).update(
-            group=group,
-            environment=environment,
-        )
+            tsdb.record_frequency_multi(frequencies, timestamp=event.datetime)
+
+        if group:
+            UserReport.objects.filter(
+                project=project,
+                event_id=event_id,
+            ).update(
+                group=group,
+                environment=environment,
+            )
```

Comment:
If we get a user-report for a group-less event it won't ever end up being displayed. We might want to account for that in how user-feedback displays.

Reply:
Yea - a follow up for sure. I think we need tweak user reports to get marked as valid once they're paired with an event.